lib/java_buildpack/framework/sealights_agent.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/framework'
+require 'java_buildpack/component/versioned_dependency_component'
+require 'fileutils'
+
+module JavaBuildpack
+  module Framework
+
+    # Encapsulates the functionality for enabling zero-touch Sealights support.
+    class SealightsAgent < JavaBuildpack::Component::VersionedDependencyComponent
+      # include JavaBuildpack::Util
+
+      def initialize(context)
+        super(context)
+        @logger = JavaBuildpack::Logging::LoggerFactory.instance.get_logger SealightsAgent
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        download_zip(false)
+        @droplet.copy_resources
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#release)
+      def release
+        @droplet.java_opts.add_javaagent(agent)
+        credentials = @application.services.find_service(FILTER, TOKEN)['credentials']
+        @droplet.java_opts.add_system_property('sl.token', credentials[TOKEN])
+        @droplet.java_opts.add_system_property('sl.tags', 'pivotal_cloud_foundry')
+        add_system_property 'sl.enableUpgrade', ENABLE_UPGRADE
+        add_system_property 'sl.buildSessionId', BUILD_SESSION_ID
```

Comment:
I don't know if it will matter here, or for any of your configuration properties, but this `add_system_property` method does *not* handle escaping of values. If a value for any of your configuration options could legitimately contain spaces or any characters that would be shell-interpreted, you'd need to shell-escape them first. 

This got us with another agent recently, you can see how to properly escape them here. https://github.com/cloudfoundry/java-buildpack/blob/main/lib/java_buildpack/framework/app_dynamics_agent.rb#L83 Basically just wrap values with `Shellwords` module & it handles the escaping.

Reply:
Thanks for pointing this out. While our tokens are URL/base64 encoded and ought to be safe, I've added this just in case.